File: services/src/Moe.La.Core/Entities/Litigation/Hearing/Hearing.cs
using Moe.La.Core.Enums;
using System;
using System.Collections.Generic;

namespace Moe.La.Core.Entities
{
    public class Hearing : BaseEntity<int>
    {
        /// <summary>
        /// The related case ID.
        /// </summary>
        public int CaseId { get; set; }

        /// <summary>
        /// The related case object.
        /// </summary>
        public Case Case { get; set; }

        /// <summary>
        /// The user ID who is responsible for attending and closing this hearing.
        /// </summary>
        public Guid? AssignedToId { get; set; }

        /// <summary>
        /// The user object who is responsible for attending and closing this hearing.
        /// </summary>
        public AppUser AssignedTo { get; set; }

        public HearingStatuses Status { get; set; }

        public HearingTypes Type { get; set; }

        public int CourtId { get; set; }

        public Court Court { get; set; }

        public string CircleNumber { get; set; }

        public int? HearingNumber { get; set; }

        public DateTime HearingDate { get; set; }

        public string HearingTime { get; set; }

        public string HearingDesc { get; set; }

        /// <summary>
        /// طلبات الجلسة
        /// </summary>
        public string Motions { get; set; }

        /// <summary>
        /// ملخص الجلسة يتم اضافتها بعد حضور الجلسة
        /// </summary>
        public string Summary { get; set; }

        /// <summary>
        /// هل تم النطق بالحكم ام لا
        /// </summary>
        public bool? IsPronouncedJudgment { get; set; }

        /// <summary>
        /// الحضور
        /// </summary>
        public string Attendees { get; set; }

        /// <summary>
        /// The hearing report.
        /// </summary>
        public string SessionMinutes { get; set; }

        /// <summary>
        /// The hearing closing report.
        /// </summary>
        public string ClosingReport { get; set; }

        /// <summary>
        /// The user's id who created it.
        /// </summary>
        public Guid CreatedBy { get; set; }

        /// <summary>
        /// Navigation property to the user who created it.
        /// </summary>
        public AppUser CreatedByUser { get; set; }

        /// <summary>
        /// The user's id who updated it.
        /// </summary>
        public Guid? UpdatedBy { get; set; }

        /// <summary>
        /// Navigation property to the user who updated it.
        /// </summary>
        public AppUser UpdatedByUser { get; set; }

        /// <summary>
        /// The update datetime.
        /// </summary>
        public DateTime? UpdatedOn { get; set; }

        public ICollection<CaseSupportingDocumentRequest> CaseSupportingDocumentRequests { get; set; } = new List<CaseSupportingDocumentRequest>();

        public ICollection<HearingLegalMemo> HearingLegalMemos { get; set; } = new List<HearingLegalMemo>();

        public ICollection<AddingLegalMemoToHearingRequest> HearingLegalMemoReviewRequests { get; set; } = new List<AddingLegalMemoToHearingRequest>();

        public ICollection<HearingAttachment> Attachments { get; set; } = new List<HearingAttachment>();
    }
}

File: website/Moe.La.Web/els-app/src/app/core/models/hearing.ts
import { CaseListItem, CaseDetails } from "./case";
import { CaseSupportingDocumentRequest } from './supporting-document-request';
import { KeyValuePairs } from './key-value-pairs';
import { Attachment } from './attachment';
import { RequestList } from './request';
import { Guid } from 'guid-typescript';
import { HearingStatus } from "../enums/HearingStatus";


export interface HearingListItem {
  id: number;
  case: CaseListItem;
  court: string;
  status: string;
  type:string;
  hearingNumber?: number;
  hearingDate: string;
  hearingTime?: string;
  hearingDesc: string;
  circleNumber: string;
  assignedTo?: KeyValuePairs<string>;
  closingReport?: string;
  deocumentRequest?: CaseSupportingDocumentRequest;
  createdOnHigri?:string;
}

export class HearingDetails {
  id: number = 0;
  case: CaseDetails = null!;
  assignedTo: KeyValuePairs<string> = { id: '', name: '' };
  court: KeyValuePairs = { id: 0, name: '' };
  status: KeyValuePairs = { id: 0, name: '' };
  type: KeyValuePairs = { id: 0, name: '' };
  hearingNumber?: number;
  hearingDate!: Date;
  hearingDateHigri: string = '';
  hearingTime: string = '';
  hearingDesc: string = '';
  motions: string = '';
  circleNumber: string = '';
  summary?: string;
  attendees?:string;
  isPronouncedJudgment?: boolean;
  //closed?: boolean;
  closingReport?: string;
  //legalMemoId: number = 0;
  //legalMemo: any;
  hearingLegalMemoReviewRequest?: HearingLegalMemoReviewRequest;
  legalMemos!:KeyValuePairs[];
  caseSupportingDocumentRequests: RequestList[]=[];
  sessionMinutes:string="";
  isEditable:boolean=true;
  isHasNextHearing:boolean;
  attachments: Attachment[]=[];
  createdOnHigri?:string;

}

export class HearingLegalMemoReviewRequest {
  Id?: number;

  HearingId: number = 0;

  LegalMemoId: number = 0;
  legalMemo?: any;
  request?:any;
  replyDate?:any;
  replyDateHigri?:string='';
  isDeleted?:any;
}


export class SaveHearing {
  id: number = 0;
  caseId: number = 0;
  courtId : number = 0
  assignedToId?: string;
  status: HearingStatus;
  type:string='';
  hearingNumber?: number;
  hearingDate?: string;
  hearingTime?: string;
  hearingDesc?: string;
  motions?: string;
  circleNumber: string = '';
  summary?: string;
  isPronouncedJudgment?: boolean;
  closingReport?: string;
  attendees?:string;
  sessionMinutes?:string;
  withNewHearing:boolean=false;
  newHearing?:SaveHearing;
  attachments?: Attachment[];
}
export interface ReceivingJudgment {
  caseId: number;
  hearingId: number;
  hearingDate: Date;
  isPronouncedJudgment?: boolean;
  pronouncingJudgmentDate?: Date;
  receivingJudgmentDate?: Date;
}

export interface SaveAndCloseHearing {
  currentHearingId: number;
  closingReport: string;
  newHearing: SaveHearing;
}

export class HearingUpdateListItem {
  id: number=0;
  hearingId?: number;
  hearing?: any;
  text: string='';
  updateDate?: Date;
  updateDateHijri:string='';
  updateTime: string='';
  CreatedBy?:KeyValuePairs<Guid>;
  attachments?: Attachment[];
  attachment?:KeyValuePairs<string>;
}

export class HearingUpdateDetails {
  id: number=0;
  hearingId?: number;
  hearing?: HearingDetails;
  text: string='';
  updateDate?: Date;
  updateDateHijri:string='';
  updateTime: string='';
  CreatedBy?:KeyValuePairs<Guid>;
  attachments?: Attachment[];
}

export class HearingUpdateDto {
  id: number=0;
  hearingId?: number;
  text: string='';
  attachments:Attachment[] = [];
}
